#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import double_t
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator
from dynamic_reconfigure.parameter_generator_catkin import str_t


PACKAGE = "dualarm_grasping"


gen = ParameterGenerator()

gen.add('score_thresh', double_t, 0, 'Score thresh', 0.5, 0.0, 1.0)
gen.add('grasp_thresh', double_t, 0, 'Grasp Score thresh', 0.3, 0.0, 1.0)
gen.add('nms_thresh', double_t, 0, 'NMS thresh', 0.3, 0.0, 1.0)
gen.add('vis_thresh', double_t, 0, 'Vis region ratio thresh', 0.8, 0.0, 1.0)
gen.add('sampling_thresh', double_t, 0, 'Sampling Score thresh', 0.7, 0.0, 1.0)
gen.add('grasping_way', str_t, 0, 'Grasping way', 'single')

exit(gen.generate(
    PACKAGE, PACKAGE, 'DualarmOccludedGraspInstanceSegmentation'))
